src/lib/FreeC/Backend/Agda/Syntax.hs

```diff
@@ -189,8 +193,11 @@ fun l             = Fun NoRange (defaultArg l)
 --
 --   > pi [α₁, …, αₙ] expr ↦ ∀ {α₁} … {αₙ} → expr
 pi :: [Name] -> Expr -> Expr
-pi decls =
-  Pi [TBind NoRange (hiddenArg <$> decls) (Underscore NoRange Nothing)]
+pi decls (Pi binder expr) = Pi
+  (TBind NoRange (hiddenArg <$> decls) (Underscore NoRange Nothing) : binder)
+  expr
+pi decls expr =
+  Pi [TBind NoRange (hiddenArg <$> decls) (Underscore NoRange Nothing)] expr
```

Comment:
Use `<$>` for computations and `map` for data.

```suggestion
pi decls (Pi binder expr) = Pi
  (TBind NoRange (map hiddenArg decls) (Underscore NoRange Nothing) : binder)
  expr
pi decls expr =
  Pi [TBind NoRange (map hiddenArg decls) (Underscore NoRange Nothing)] expr
```

This can probably also be rewritten such that there is less code duplication.